File: sdk/rust/examples/export.rs
use oomrpc::Client;
use std::time::{SystemTime, UNIX_EPOCH};

macro_rules! svec { ($($part:expr),* $(,)?) => {{ vec![$($part.to_string(),)*] }} }

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut client = Client::connect("http://127.0.0.1:50051").await?;

    let features = svec![
        "account.state",
        "account.has_2fa_installed",
        "transaction_stats.transaction_count_7d"
    ];

    let now = SystemTime::now().duration_since(UNIX_EPOCH)?.as_millis().try_into()?;
    client.export(features, now, "/tmp/output.csv", 20).await?;

    std::fs::copy("/tmp/output.csv", "/dev/stdout")?;

    Ok(())
}

File: sdk/python_v2/examples/export.py
#!/usr/bin/env python3
import asyncio
import time
from oomclient import Client


async def main():
    client = await Client.connect("http://localhost:50051")
    features = [
        "account.state",
        "account.has_2fa_installed",
        "transaction_stats.transaction_count_7d",
    ]
    output_path = "/tmp/output.csv"
    now = round(time.time() * 1000)
    await client.export(features, now, output_path, 20)
    print(open(output_path).read())


asyncio.run(main())
